pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -524,11 +557,20 @@ func NewOpenStackClient(cfg *AuthOpts, userAgent string, extraUserAgent ...strin
 		}
 	}
 
+	config := &tls.Config{}
+	config.InsecureSkipVerify = cfg.TLSInsecure == "true"
+
 	if caPool != nil {
-		config := &tls.Config{}
 		config.RootCAs = caPool
-		config.InsecureSkipVerify = cfg.TLSInsecure == "true"
-		provider.HTTPClient.Transport = netutil.SetOldTransportDefaults(&http.Transport{TLSClientConfig: config})
+	}
+
+	provider.HTTPClient.Transport = netutil.SetOldTransportDefaults(&http.Transport{TLSClientConfig: config})
```

Comment:
@kayrus I see you have done same changes in other PR as well , Could you address comments there or merge PR to avoid multiple reviews of same thing.

Reply:
I addressed this in https://github.com/kubernetes/cloud-provider-openstack/pull/867#pullrequestreview-327561280 comment. It should be find to use tls.Config, when plain HTTP is used.